hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1519,7 +1521,7 @@ private void listStatusFindKeyInTableCache(
 
       // First, find key in TableCache
       listStatusFindKeyInTableCache(cacheIter, keyArgs, startCacheKey,
```

Comment:
Can we initialize `int countEntries = cacheKeyMap.size();` instead of  `int countEntries = 0;` ? 
This will ensure only the delta will be fetched from OM DB.

Also, I think `map -> to -> list `conversion logic can be simplified because at the end `cacheKeyMap.size() <= numEntries`.
[KeyManagerImpl.java#L1596](https://github.com/apache/ozone/blob/master/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java#L1596)

Reply:
Do we need to preserve the sorted enumeration?

Reply:
Agreed.  This logic is needed to preserve the sorted order.

_For example:_
Assume there are CachedKeys = 1000.
Now the list logic will fetch complete 1024 numEntries from om DB. Then it combines and do sorting. In worst case, it will pickup all 1000 entries from cache but only use 24 entries from om DB, which resulted in 1000 unnecessary lookup on DB. In another case, it can happen like will use only a few entries from cache and use a few keys from om DB.